### PR TITLE
tested few cases

### PR DESCRIPTION
**Pull request by MrishoLukamba**
Tested some functions and edge cases works fine , The error messages are not friendly yet.

### CODE REVIEW

File: pallets/roles/src/tests.rs [hunk 1]
```diff
@@ -1,20 +1,41 @@
...
+				share:0,
+				selections:0,
+			}
+		);
+		//---checking storage----------------
+		assert_eq!(InvestorLog::<Test>::contains_key(1),true);
+
+		//--trying creating same investor account,it should fail
+		assert_noop!(RoleModule::create_account(Origin::signed(1),Accounts::INVESTOR),
+			Error::<Test>::NoneValue
```
Comment (ilhanu) [inline pallets/roles/src/tests.rs:29]: Hey Mrisho, thanks for being the only one testing 😅 You are early as the functionality is just being added.

It would be nice if the test fails it reports the error in the pallet

Please see the remainder of the tests:
- [x]  Getting roles, successful
- [ ]  Getting several roles, unsuccessful  **This needs more extensive testing.**
- [ ]  Getting role (seller, servicer) without verification, unsuccessful

These are just examples you are also welcome to code your own test you think that are valuable.

**Review by ilhanu — CHANGES_REQUESTED** [review]
Thanks for adding some tests, the PR is bit early, but some early tests help to further kickstart it. The branch is not checking though, try to merge stuff after the merge of Kazu's branch within your own branch. It will most likely fix it.

File: pallets/roles/src/structs.rs [hunk 2]
```diff
@@ -85,7 +83,7 @@ impl<T:Config> Investor<T> where structs::Investor<T>: EncodeLike<structs::Inves
 pub struct HouseSeller<T: Config>{
     pub account_id:T::AccountId,
     pub nft_index:Vec<u32>,
-    pub age:BlockNumberOf<T>,
+    pub age:T::BlockNumber,
```
[the remark this answers is not in the document]
Comment (MrishoLukamba) [reply pallets/roles/src/structs.rs:86]: No it doesn't bring conflicts. So the type aliases was not needed as all associated types were on config trait. And i think code consistency is good

File: pallets/roles/src/tests.rs (hunk 1 above)
Comment (MrishoLukamba) [reply pallets/roles/src/tests.rs:29]: Okey

File: .idea/vcs.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="" vcs="Git" />
+  </component>
+</project>
```
[the remark this answers is not in the document]
Comment (MrishoLukamba) [reply .idea/vcs.xml:6]: Will do

**Comment by MrishoLukamba** [comment]
So i should wait clone the kazu branch repo and do testing and he will submit as a whole tested branch ryt

**Review by ndkazu — COMMENTED** [review]
The roles pallet has been finalized and merged in main. This is creating some conflicts in your PR, and we should discuss as a team about the way to deal with this (merge pull?).
